zingolib/src/wallet/keys/unified.rs

```diff
@@ -84,53 +116,82 @@ fn read_write_receiver_selections() {
     }
 }
 
-impl UnifiedSpendCapability {
+impl WalletCapability {
     pub fn addresses(&self) -> &[UnifiedAddress] {
         &self.addresses
     }
 
-    pub fn transparent_child_keys(&self) -> &Vec<(usize, secp256k1::SecretKey)> {
-        &self.transparent_child_keys
+    pub fn transparent_child_keys(&self) -> Result<&Vec<(usize, secp256k1::SecretKey)>, String> {
+        if self.transparent.can_spend() {
+            Ok(&self.transparent_child_keys)
+        } else {
+            Err("Wallet is not capable to spend transparent funds.".to_string())
+        }
     }
 
     pub fn new_address(
         &mut self,
         desired_receivers: ReceiverSelection,
     ) -> Result<UnifiedAddress, String> {
+        if (desired_receivers.transparent & !self.transparent.can_view())
+            | (desired_receivers.sapling & !self.sapling.can_view()
+                | (desired_receivers.orchard & !self.orchard.can_view()))
+        {
+            return Err("Wallet is no capable to produce desired receivers".to_string());
```

Comment:
"Wallet is not capable of producing desired receivers."

Reply:
3803f14f23f0bbf7229ca7878ffbdd86fba887db